File: 01-Languages/CSharp/22-Mini-Projects/TaskTracker/TaskNotFoundException.cs
namespace TaskTracker;

// A dedicated exception type (rather than a generic InvalidOperationException)
// lets the CLI layer catch exactly this failure mode and print a clean
// user-facing message, without accidentally swallowing unrelated bugs that
// also happen to throw InvalidOperationException elsewhere in the repository.
public class TaskNotFoundException : Exception {
    public int TaskId { get; }

    public TaskNotFoundException(int taskId)
        : base($"No task found with id {taskId}.") {
        TaskId = taskId;
    }
}

File: 01-Languages/PHP/22-Mini-Projects/TaskTracker/src/TaskNotFoundException.php
<?php

declare(strict_types=1);

/**
 * A custom, specific exception rather than a generic \RuntimeException --
 * a caller catching TaskNotFoundException by name knows exactly what went
 * wrong without inspecting the message string, and CLI code can catch just
 * this type without also swallowing unrelated runtime failures.
 */
final class TaskNotFoundException extends RuntimeException
{
    public function __construct(int $id)
    {
        parent::__construct("No task found with id {$id}.");
    }
}
